amida-tech/indaba-client
# Feature/INBA-612 File Upload

#### What does this PR do?
Allows respondents to upload and associate a file with an answer for questions that have the file metadata flag.

#### Related JIRA tickets:
[INBA-612](https://jira.amida-tech.com/browse/INBA-612)

#### How should this be manually tested?
1. Create a project with a survey with one or more questions with the file checkbox checked.
1. Create 4 stages, one of each type
1. Assign users to a subject for all stages
1. Sign in as the assigned user and load the tasks in order
1. Check that a file upload form is rendered for `Complete` and `Review and Edit` stages

For `Complete` and `Review and Edit` stages
1. Check that you can input a file name in the form, and select a file by clicking the Choose File button
1. Check that clicking Add File creates an entry in the `file` table in the survey service, and adds values to the `meta` column of the corresponding `answer` row with the `id` and `filename`
1. Check that the form now displays the filename and a Remove File button
1. Check that the form remains that way after a refresh
1. Check that clicking the Remove File button removes the meta values from the answer row (there is currently no end point for deleting the `file` entry)

#### Background/Context

#### Screenshots (if appropriate):
